### PR TITLE
Additional 2m P12 changes.

### PR DESCRIPTION
**Pull request by Gampyg28**

### CODE REVIEW

**Comment by antuspcm** [comment]
I think this might be the wrong approach. We need to validate if the top extra 1Mb is used. Its possible (likely?) that it is like the P10 we still want to set image size to 1Mb for this pcm and leave everything else as it is depending on if that the top part of the chip is not addressable or is empty. Im testing this at the moment to see which approach is correct.

**Comment by Gampyg28** [comment]
That is the EXACT reason for the hack in Write_Background to disable Writes on it.

We just don't know until we get a read.

File: Apps/PcmHammer/MainForm.cs [hunk 1]
```diff
@@ -1451,6 +1451,13 @@ private async void write_BackgroundThread(WriteType writeType, string path = nul
                         }
                     }
 
+                    // Temporary hack to disable write access for the 2m P12 until further testing has been accomplished.
```
Comment (Gampyg28) [inline Apps/PcmHammer/MainForm.cs:1454]: The exact reason for this ...

We just don't know until we get a read.

**Comment by LegacyNsfw** [comment]
Looks good to me, shall I merge it?

**Review by LegacyNsfw — APPROVED** [review]

**Comment by antuspcm** [comment]
I think not. Darkman5001 is testing and this version should attempt to give us a full read of a 2mb chip. So far it bombs out at another pre-flight check - the flash chip being a 512k or 1mb image, so he'll be testing again soon next time he has the car. When we have the answer we can modify this branch to either allow write for 2mb chips, or revert this and configure pcmhammer to only use the first 1mb. This is good progress, but IMHO shouldnt go in to pcmhammer develop.

**Comment by antuspcm** [comment]
Ku4ro also said the file format is different, so this might reject the bin and not save it. We can skip that check and try for a 3rd time if need be. I would also like to identify if these are the P12B PCMs that are out there.

**Comment by plastic-racer** [comment]
I am awaiting approval to join pcmhacking forum, but have quite a few bins (including slave and shadow) including a couple from the supposed 2mb pcm OSs listed

**Comment by antuspcm** [comment]
> I am awaiting approval to join pcmhacking forum, but have quite a few bins (including slave and shadow) including a couple from the supposed 2mb pcm OSs listed

Please reply to the email you received on signup (as stated in its last line), It's a necessary anti-spam measure.

**Comment by plastic-racer** [comment]
> > I am awaiting approval to join pcmhacking forum, but have quite a few bins (including slave and shadow) including a couple from the supposed 2mb pcm OSs listed
> 
> Please reply to the email you received on signup (as stated in its last line), It's a necessary anti-spam measure.

I replied to the email, once I can get in I will reply to the thread and attach the bins (i have several OSs including the assumed 2mb ones)

**Comment by antuspcm** [comment]
> > > I am awaiting approval to join pcmhacking forum, but have quite a few bins (including slave and shadow) including a couple from the supposed 2mb pcm OSs listed
> > 
> > 
> > Please reply to the email you received on signup (as stated in its last line), It's a necessary anti-spam measure.
> 
> I replied to the email, once I can get in I will reply to the thread and attach the bins (i have several OSs including the assumed 2mb ones)

I believe I have activated you, but I am not sure who is who. If you still need activation you can email me antus at pcmhacking dot net

**Pull request merged**